sklearn/model_selection/_search_successive_halving.py

```diff
@@ -199,6 +220,15 @@ def fit(self, X, y=None, groups=None, **fit_params):
         **fit_params : dict of string -> object
             Parameters passed to the ``fit`` method of the estimator
         """
+        # y needs to be validated for the parameters validation later on
```

Comment:
does that mean `y` will be validated twice?

Also for my own understanding, what part of the parameter validation forces us to validate `y` here?

Reply:
Yes, it will be also validated by the underlying estimator.
It seems that you need to know the number of classes to decide the number of minimum resources sometimes. Thus, you need to make a `np.unique(y)`. Thus `y` needs to be an array-like and validated.

Reply:
I see, thanks.

Would it make more sense to do this validation just before the `np.unique` call then? This will make sure we only do these extra validation when we need it.

Also, are all of them needed? I would assume that `column_or_1d` could be enough here, but maybe I'm wrong